docs/concepts/components/children.md

```diff
@@ -0,0 +1,214 @@
+			</div>
+		}
+	}
+}
+```
+
+## Advanced usage
+
+Here are two cases where you would want to break away from using `Children`
+and instead use something else:
+
+ - You want to restrict the types of components that can be used as children
+   on this component.
+
+ - You want to share state between a component and its children.
+
+### Typed children
+
+In cases where you only need to restrict the children to a single component,
+you can use `yew::html::ChildrenWithProps<T>`.
+
+```rust
+use yew::prelude::*;
+use yew::html::ChildrenWithProps;
+
+// ...
+
+#[derive(Properties, Clone)]
+pub struct ListProps {
+	#[prop_or_default]
+	pub children: ChildrenWithProps<Item>,
+}
+
+pub struct List {
+	props: ListProps,
+}
+
+impl Component for ListProps {
+	type Properties = ListProps;
+	// ...
+
+	fn view(&self) -> Html {
+		html! {
+			<div class="list">
+				{ for self.props.children.iter() }
+			</div>
+		}
+	}
+}
+```
+
+Of course, sometimes you might need to restrict the children to a few different
+components. In these cases, you have to get a little more hands-on with yew.
+
+The `derive_more` crate is used here for better ergonomics. If you don't want
+to use it, replace `ItemPropVariants` with this:
+
+```rust
+#[derive(Clone)]
+pub enum ItemPropVariants {
+	MyFirstComponent(MyFirstComponentProps),
+	MySecondComponent(MySecondComponentProps),
+}
+
+impl From<MyFirstComponentProps> for ItemPropVariants {
+	fn from(props: MyFirstComponentProps) -> Self {
+		Self::MyFirstComponent(props)		
+	}
+}
+
+impl From<MySecondComponentProps> for ItemPropVariants {
+	fn from(props: MySecondComponentProps) -> Self {
+		Self::MySecondComponent(props)
+	}
+}
+```
+
+```rust
+use yew::prelude::*;
+use yew::html::ChildrenRenderer;
+use yew::virtual_dom::{ VChild, VComp };
+
+// ...
+
+#[derive(Clone, derive_more::From)]
+pub enum ItemPropVariants {
+	MyFirstComponent(MyFirstComponentProps),
+	MySecondComponent(MySecondComponentProps),
+}
+
+#[derive(Clone)]
+pub struct Item {
+	props: ItemPropVariants,
+}
+
+impl<CHILD> From<VChild<CHILD>> for Item
+where
+	CHILD: Component,
+	CHILD::Properties: Into<ItemPropVariants>,
+{
+	fn from(vchild: VChild<CHILD>) -> Self {
+		Item {
+			props: vchild.props.into(),
+		}
+	}
+}
+
+impl Into<Html> for Item {
+	fn into(self) -> Html {
+		match self.props {
+			ItemPropVariants::MyFirstComponent(props) => {
+				VComp::new::<MyFirstComponent>(props, NodeRef::default(), None).into()
+			}
+			ItemPropVariants::MySecondComponent(props) => {
+				VComp::new::<MySecondComponent>(props, NodeRef::default(), None).into()
+			}
+		}
+	}
+}
+
+#[derive(Properties, Clone)]
+pub struct ListProps {
+	#[prop_or_default]
+	pub children: ChildrenRenderer<Item>,
+}
+
+pub struct List {
+	props: ListProps,
+}
+
+impl Component for List {
+	type Properties = ListProps;
+	// ...
+
+	fn view(&self) -> Html {
+		html! {
+			<div class="list">
+				{ for self.props.children.iter() }
+			</div>
+		}
+	}
+}
+```
```

Comment:
I feel like a little more explanation would be good here.

Reply:
Will do!

Reply:
@teymour-aldridge I've made my first draft of an explanation in [this](https://github.com/yewstack/yew/pull/1616/commits/f02e0448a9974056d1c762bb25c65c2a94c822e3) commit. I'm relatively new-ish to Yew, so some of it might not be 100% accurate, but I think its right for the most part.

Reply:
I think the explanation is great, but I think most of it could be inlined as comments. That way you get the explanation while reading the code

Reply:
:+1: Changed in commit [405ea75](https://github.com/yewstack/yew/pull/1616/commits/405ea755d50c4153c1c890fc9fc273456b8e4f76)